src/engine/positionscratchcontroller.cpp

```diff
@@ -321,22 +308,25 @@ void PositionScratchController::process(double currentSamplePos,
         m_samplePosDeltaSum = -(releaseRate / m_p) * m_callsPerDt;
         m_pVelocityController->reset(-m_samplePosDeltaSum);
         m_pRateIIFilter->reset(-m_samplePosDeltaSum);
-        m_scratchStartPos = scratchPosition;
+        // The "scratch_position" CO contains relative traveled audio frames * 2
+        // Not necessarily the actual position in the track. We use this to calculate
+        // the traveled distance of the mouse compared to m_scratchStartPos.
+        m_scratchStartPos = m_pScratchPos->get();
+        m_scratchPosSampleTime = 0;
         // qDebug() << "scratchEnable()" << currentSamplePos;
     }
+
+    if (!util_isnan(m_seekSamplePos)) {
+        // We need to transpose all buffers to compensate the seek
+        // in a way that the next process call does not even notice anything
+        currentSamplePos = m_seekSamplePos;
+        m_seekSamplePos = std::numeric_limits<double>::quiet_NaN();
+    }
     m_prevSamplePos = currentSamplePos;
 }
 
 void PositionScratchController::notifySeek(mixxx::audio::FramePos position) {
-    DEBUG_ASSERT(position.isValid());
     const double newPos = position.toEngineSamplePos();
-    if (!isEnabled()) {
-        // not scratching, ignore";
-        return;
-    } else if (m_prevSamplePos == newPos) {
```

Comment:
If I restore this all is good again.
Not sure about the no-op check below, I think I kept it from the state where identical seek notifications were received due to the call from ReadAheadManager.

I'll pick your commit but edit this.

Thanks for taking a look!